__tests__/Unit/Components/ProgressForm/ProgressForm.test.tsx

```diff
@@ -81,21 +98,34 @@ describe('Progress form', function () {
         fireEvent.change(textAreas[2], { target: { value: '567' } });
 
         expect(button.className).toBe('buttonEnabled');
+        fireEvent.click(button);
+        await waitFor(() =>
+            expect(
+                screen.getByText('Task Progress saved successfully')
+            ).toBeInTheDocument()
+        );
     });
 
-    it('Check if onClick is working', function () {
+    it('onClick should not work in case of no inputs', async function () {
         renderWithRouter(
             <Provider store={store()}>
                 <ProgressForm questions={questions} />
-            </Provider>
+            </Provider>,
+            {
+                asPath: '/progress',
+                replace: jest.fn(),
+            }
         );
 
         const button = screen.getByRole('button');
-
-        button.onclick = mockOnClick;
         fireEvent.click(button);
+        expect(button.className).toBe('buttonDisabled');
```

Comment:
For betterment we can use `toHaveClass()` function

Reply:
Have updated it, Can you please check?